visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.reducer.ts

```diff
@@ -0,0 +1,118 @@
+import { EdgeMetricDataAction, EdgeMetricDataActions, setEdgeMetricData } from "./edgeMetricData.actions"
+import { BlacklistItem, BlacklistType, Edge, EdgeMetricCount, EdgeMetricData } from "../../../../codeCharta.model"
+import { getVisibleFileStates } from "../../../../model/files/files.helper"
+import { CodeMapHelper } from "../../../../util/codeMapHelper"
+import { FileState } from "../../../../model/files/files"
+import { EdgeMetricDataService } from "./edgeMetricData.service"
+
+const clone = require("rfdc")()
+
+// Required for performance improvements
+export type NodeEdgeMetricsMap = Map<string, Map<string, EdgeMetricCount>>
+export let nodeEdgeMetricsMap: NodeEdgeMetricsMap
```

Comment:
```suggestion
export let nodeEdgeMetricsMap: EdgeMetricCount = new Map()
```

I would always initialize the map directly.